src/app/components/team/Similarity/SimilarityComponent.js

```diff
@@ -80,9 +99,12 @@ const SimilarityComponent = ({
 
   const handleVectorModelToggle = (useVectorModel) => {
     if (!useVectorModel) {
-      handleSettingsChange('text_similarity_model', ELASTICSEARCH_MODEL);
+      const newSettings = { ...settings };
+      newSettings.alegre_model_in_use = ELASTICSEARCH_MODEL;
+      newSettings.text_similarity_model = ELASTICSEARCH_MODEL;
+      setSettings(newSettings);
     } else {
-      handleSettingsChange('text_similarity_model', MEAN_TOKENS_MODEL);
+      handleModelSettingsChange(MEAN_TOKENS_MODEL);
```

Comment:
When vector models are first enabled, we want `alegre_model_in_use` and `text_similarity_model` to be Arrays  including both `ELASTICSEARCH_MODEL` and `MEAN_TOKENS_MODEL`. 

I have no idea why this code:
```
      handleSettingsChange('alegre_model_in_use', [ELASTICSEARCH_MODEL, MEAN_TOKENS_MODEL]);
      handleSettingsChange('text_similarity_model', [ELASTICSEARCH_MODEL, MEAN_TOKENS_MODEL]);
```
doesn't do that. It would leave `text_similarity_model` as only being elasticsearch. In any case, `handleModelSettingsChange(MEAN_TOKENS_MODEL);` performs what is expected and the settings sent to Check API are correct.